### PR TITLE
Resurrect code coverage in Common

### PR DESCRIPTION
**Pull request by paulkagiri**
- This PR incorporates the code coverage work done on this PR - https://github.com/AzureAD/android-complete/pull/110 in Common and Common4j
- It does this by updating the buildsystem plugin within the code coverage tool is housed in and purging the manual jacoco integration that had been done before
- The reason there are 2 pipelines (one for Github Actions and the other for Azure pipelines) is because we are currently running using Github Actions but in the near future we will need to have a nightly run hence the Azure pipeline.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,5 +1,6 @@
 pluginManagement {
     repositories {
+        mavenLocal()
```
Comment (AdamBJohnsonx) [inline settings.gradle:3]: Do we need maven local here?

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          java-version: 1.8
+
+      - name: Run Coverage
+        run: ./gradlew common:localDebugCommonUnitTestCoverageReport -Plabtest -ProbolectricSdkVersion=$ROBOELECTRIC_SDK_VERSION
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          file: ${{ github.workspace }}/common/build/reports/jacoco/localDebugCommonUnitTestCoverageReport/localDebugCommonUnitTestCoverageReport.xml
```
Comment (AdamBJohnsonx) [inline .github/workflows/codecov.yml:25]:
```suggestion
          file: ${{ github.workspace }}/common/build/reports/jacoco/localDebugCommonUnitTestCoverageReport/localDebugCommonUnitTestCoverageReport.xml
          
```

File: azure-pipelines/code-coverage/common.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      publishJUnitResults: false
+  - task: Gradle@2
+    displayName: Run common code coverage
+    inputs:
+      tasks: common:localDebugCommonUnitTestCoverageReport -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}}
+  - task: PublishCodeCoverageResults@1
+    displayName: 'Publish code coverage report'
+    inputs:
+      summaryFileLocation: '$(Build.SourcesDirectory)\common\build\reports\jacoco\localDebugCommonUnitTestCoverageReport\localDebugCommonUnitTestCoverageReport.xml'
+      reportDirectory: '$(Build.SourcesDirectory)\common\build\reports\jacoco\localDebugCommonUnitTestCoverageReport\html'
```
Comment (AdamBJohnsonx) [inline azure-pipelines/code-coverage/common.yml:46]:
```suggestion
      reportDirectory: '$(Build.SourcesDirectory)\common\build\reports\jacoco\localDebugCommonUnitTestCoverageReport\html'

```

File: .github/workflows/codecov.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```
Comment (AdamBJohnsonx) [inline .github/workflows/codecov.yml:17]: Can we change this to 11 at some point very soon?

**Review by AdamBJohnsonx — APPROVED** [review]
Minor comments, mainly.  Do we need maven local at all?

**Comment by paulkagiri** [comment]
> Minor comments, mainly. Do we need maven local at all?

This has already been purged

File: azure-pipelines/code-coverage/common.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# File: azure-pipelines\code-coverage\common.yml
+# Description: Code coverage for common
+# Variable: 'mvnAccessToken' was defined in the Variables tab
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'commonBranchName' common branch to be used when running the pipeline manually
+name: $(date:yyyyMMdd)$(rev:.r)
+
+trigger: none
```
Comment (p3dr0rv) [inline azure-pipelines/code-coverage/common.yml:8]: Do we want both (GitHub Action and Azure ) pipelines to run on PR?

**Review by p3dr0rv — APPROVED** [review]

**Comment by shahzaibj** [comment]
Why is coverage being collected in 2 different places? Github Action as well as Azure Pipelines?

Also can we get one combined report with coverage report for both unit as well as instrumented tests?

File: azure-pipelines/code-coverage/common.yml (hunk 5 above)
Comment (paulkagiri) [reply azure-pipelines/code-coverage/common.yml:8]: Not at the moment - this will be done in the near future when we have a nightly run

**Comment by paulkagiri** [comment]
> Why is coverage being collected in 2 different places? Github Action as well as Azure Pipelines?
> 
> Also can we get one combined report with coverage report for both unit as well as instrumented tests?

@shahzaibj I have documented this on the PR description, the combined one will be done in the near future

**Pull request merged**